dynamic_programming/house_robber.cpp

```diff
@@ -0,0 +1,95 @@
+/*House Robber - Dynamic Programming
+
+Problem Statement: 
+A robber is planning to rob houses along a street. Each house has a 
+certain  amount  of money stashed, the only constraint stopping you 
+from  robbing  each of them  is that  adjacent houses have security 
+system connected and it will automatically contact the police if two
+adjacent houses were broken into on the same night.
+
+Task: 
+Determine the maximum amount of money you can rob tonight without alerting the police
+
+Solution: 
+For every house we have two options either to rob it, if we do so we have to
+move to next to next huse as we cannot rob two adjacent houses. If we don't rob this
+house then we can move to the next one. The maximum of above two cases will be the maximum profit.
+We store our preiously calculated results in a table (dpTable) if we found that we have already
+calculated the ans then we simply return it.
+*/
+
+
+#include <iostream>
+#include <vector> 
+#include <cassert>
+
+/**
+ * @brief This function implements the above algorithm
+ * @money array of numbers containing the money in the ith house
```

Comment:
the doc blocks are not per the requested, suggested and described in the guidelines. refer to the comment above.

The "This" does not reflect in the documentation. You can review the documentation site: http://thealgorithms.github.io/C-Plus-Plus
You can split the file and function documentations. See [this file](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/dynamic_programming/kadane2.cpp) for example. Also note the use of namespaces.